File: backend/src/main/java/app/miniappspring/controller/DiscountController.java
package app.miniappspring.controller;

import app.miniappspring.dto.discount.DiscountCreateDto;
import app.miniappspring.dto.discount.DiscountDto;
import app.miniappspring.service.DiscountService;
import lombok.RequiredArgsConstructor;
import org.springframework.security.access.prepost.PreAuthorize;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@RequestMapping("/api/v1/discounts")
@RequiredArgsConstructor
public class DiscountController {
    private final DiscountService discountService;

    @GetMapping("")
    public List<DiscountDto> getDiscountList() {
        return discountService.getDiscountList();
    }

    @PostMapping("/add")
    @PreAuthorize("hasAnyAuthority('ROLE_EMPLOYEE')")
    public DiscountDto addDiscount(@RequestBody DiscountCreateDto discountCreateDto){
        return discountService.createDiscount(discountCreateDto);
    }

    @GetMapping("/check")
    public DiscountDto checkDiscount(@RequestParam Long productId){
        return discountService.checkDiscountAtProduct(productId);
    }

}

File: frontend/react-app/src/controller/DiscountController.ts
import api from "../http";
import {IDiscountCreate} from "../model/discount/IDiscountCreate";
import {IDiscount} from "../model/discount/IDiscount";
import {ICardProduct} from "../model/product/ICardProduct";

export class DiscountController {
    static async checkDiscountAtProduct(productId: string) {
        return api.get<IDiscount>(`/api/v1/discounts/check?productId=${productId}`).then(response=>response.data);
    }

    static async createDiscount(discount:IDiscountCreate) {
       return api.post<IDiscount>('/api/v1/discounts/add',discount).then(response=>response.data);
    }

    static getProductWithDiscountByTown(town: string) {
        return api.get<ICardProduct[]>(`/api/v1/discounts/products?town=${town}`, {}).then(response => response.data);
    }
}